varats/varats/projects/perf_tests/feature_perf_regression.py

```diff
@@ -0,0 +1,112 @@
+"""Project file for the feature performance case study collection."""
+import re
+import typing as tp
+
+import benchbuild as bb
+from benchbuild.command import Command, SourceRoot, WorkloadSet
+from benchbuild.utils.cmd import make, cmake, mkdir
+from benchbuild.utils.settings import get_number_of_jobs
+from plumbum import local
+
+from varats.experiment.workload_util import RSBinary, WorkloadCategory
+from varats.paper.paper_config import PaperConfigSpecificGit
+from varats.project.project_domain import ProjectDomains
+from varats.project.project_util import (
+    ProjectBinaryWrapper,
+    BinaryType,
+    get_local_project_git_path,
+    verify_binaries,
+    get_tagged_commits,
+)
+from varats.project.sources import FeatureSource
+from varats.project.varats_project import VProject
+from varats.provider.release.release_provider import (
+    ReleaseProviderHook,
+    ReleaseType,
+)
+from varats.utils.git_util import (
+    RevisionBinaryMap,
+    ShortCommitHash,
+    FullCommitHash,
+)
+from varats.utils.settings import bb_cfg
+
+
+class FeaturePerfRegression(VProject, ReleaseProviderHook):
+    """Test project for identifying performance regressions across revisions."""
+
+    NAME = 'FeaturePerfRegression'
+    GROUP = 'perf_tests'
+    DOMAIN = ProjectDomains.TEST
+
+    SOURCE = [
+        PaperConfigSpecificGit(
+            project_name="FeaturePerfRegression",
+            remote="https://github.com/bnico99/FeaturePerfRegression.git",
+            local="FeaturePerfRegression",
+            refspec="origin/HEAD",
+            limit=None,
+            shallow=False
+        ),
+        FeatureSource()
+    ]
+
+    WORKLOADS = {
+        WorkloadSet(WorkloadCategory.EXAMPLE): [
+            Command(
+                SourceRoot("FeaturePerfRegression") /
+                RSBinary("CompressionTool"),
+                label="FeaturePerfRegression-no-input"
+            )
+        ]
+    }
+
+    @staticmethod
+    def binaries_for_revision(
+        revision: ShortCommitHash
+    ) -> tp.List[ProjectBinaryWrapper]:
+        binary_map = RevisionBinaryMap(
+            get_local_project_git_path(FeaturePerfRegression.NAME)
+        )
+        binary_map.specify_binary(
+            'build/bin/CompressionTool', BinaryType.EXECUTABLE
+        )
+
+        return binary_map[revision]
+
+    def run_tests(self) -> None:
+        pass
+
+    def compile(self) -> None:
+        """Compile the project."""
+        feature_perf_regression_source = local.path(
+            self.source_of(self.primary_source)
+        )
+
+        # Decrease instruction threshold and disable exceptions
+        self.cflags += ["-fvara-instruction-threshold=1", "-fno-exceptions"]
```

Comment:
```suggestion
        self.cflags += ["-fvara-instruction-threshold=1", "-fno-exceptions"]
```
The threshold is experiment specific for feature instr. and should not be part of the project.

I'm not sure what the best solution is here :thinking:

Reply:
Maybe the threshold could be added as a command line option that is passed to the experiment or incorporated into the case_study file somehow, if that makes sense to you?

Reply:
The case_study file is a bit to generic for that and would then need some form of custom section :thinking: 

@Kaufi-Jonas at least added a variable way to pass the threshold to the base class for correctly setting it in the cflags, however, this still does not solve the problem of choosing a correct value. From the conceptual point of view, it's the job of the experiment to define "how fine grained" measurements should be made. So, the problem is that we currently don't have a good way for the experiment to determine this. One idea I currently have floating in my head would be to use the 'Domain' of the project. If it's a test project set it to 0/1 if not use the default. For more elaborate work one could come up with a better strategy to select the threshold, however, that would also require some form of project introspection to determine it.